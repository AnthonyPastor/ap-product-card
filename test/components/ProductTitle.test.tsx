import React from 'react';
import renderer from 'react-test-renderer';

import { ProductTitle, ProductCard } from '../../src';
import { product1, product2 } from '../data/products';

describe('pruebas en <ProductTitle/>', () => {
  test('debe de mostrar el componenete con el titutlo personalizado', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom Product" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar el componenete con el titulo de producto 1', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar el componenete con el titulo de producto 2', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductTitle />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
