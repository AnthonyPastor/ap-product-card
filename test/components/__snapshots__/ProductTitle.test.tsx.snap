// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <ProductTitle/> debe de mostrar el componenete con el titulo de producto 1 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee mug - Sin imagen
  </span>
</div>
`;

exports[`pruebas en <ProductTitle/> debe de mostrar el componenete con el titulo de producto 2 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee mug - Con imagen
  </span>
</div>
`;

exports[`pruebas en <ProductTitle/> debe de mostrar el componenete con el titutlo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
